hs_core/search_indexes.py

```diff
@@ -44,65 +48,130 @@ def get_model(self):
         return BaseResource
 
     def index_queryset(self, using=None):
-        return self.get_model().objects.all()
+        return self.get_model().objects.filter(Q(raccess__discoverable=True) | Q(raccess__public=True))
 
     def prepare_title(self, obj):
-        return obj.metadata.title.value
+        if hasattr(obj.metadata.title, 'value'):
+            return obj.metadata.title.value
+        else: 
+            logger.debug("resource object does not have metadata field")
+            return 'none'
 
     def prepare_creators(self, obj):
-         return [creator.name for creator in obj.metadata.creators.all()]
+        if hasattr(obj, 'metadata'): 
+            return [creator.name for creator in obj.metadata.creators.all()]
+        else: 
+            logger.debug("resource object does not have metadata field")
```

Comment:
@xuezhaokun remove the debug log as well as other similar debug logs in subsequent prepare_ methods.